File: 0013-roman-to-integer/0013-roman-to-integer.swift
class Solution {
    func romanToInt(_ s: String) -> Int {
        var conversion = ["I": 1, "V": 5, "X": 10, "L": 50, "C": 100, "D": 500, "M": 1000]
        let string = Array(s)
        var intNumber = 0
        var previous = ""
        for romanNumber in string {
            let roman = String(romanNumber)
            intNumber += conversion[roman] ?? 0
            if previous == "I" && (roman == "V" || roman == "X") {
                intNumber -= 2
            }
            else if previous == "X" && (roman == "L" || roman == "C") {
                intNumber -= 20
            }
            else if previous == "C" && (roman == "D" || roman == "M") {
                intNumber -= 200
            }
            previous = roman
        }
        return intNumber
    }
}

File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        roman_values = {
            'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M':1000 
        }
        result = 0
        i = 0
        while (i < len(s)-1):
            if roman_values[s[i]] < roman_values[s[i+1]]:
                result += roman_values[s[i+1]] - roman_values[s[i]]
                i += 2
            else:
                result += roman_values[s[i]]
                i += 1  
        if i < len(s):
            result += roman_values[s[-1]]
        return result